assets/js/app.js

```diff
@@ -10,70 +10,137 @@ import css from "../css/app.css"
 // Import dependencies
 //
 import "phoenix_html"
+import { Presence, Socket } from "phoenix"
+import * as monaco from "monaco-editor"
 
 // Import local files
 //
 // Local files can be imported directly using relative paths, for example:
 // import socket from "./socket"
 
-import * as monaco from 'monaco-editor';
 
-console.log('hello1')
+class MonacoEditor {
+  editor(userColor) {
+    // Since packaging is done by you, you need
+    // to instruct the editor how you named the
+    // bundles that contain the web workers.
+    self.MonacoEnvironment = {
+      getWorkerUrl: function (moduleId, label) {
+        if (label === 'json') {
+          return './js/json.worker.js';
+        }
+        if (label === 'css') {
+          return './js/css.worker.js';
+        }
+        if (label === 'html') {
+          return './js/html.worker.js';
+        }
+        if (label === 'typescript' || label === 'javascript') {
+          return './js/ts.worker.js';
+        }
+        return './js/editor.worker.js';
+      }
+    }
 
-// var editor = monaco.editor.create(document.getElementById('container'), {
-//   value: [
-//     'function x() {',
-//     '\tconsole.log("Hello world!");',
-//     '}'
-//   ].join('\n'),
-//   language: 'javascript',
-//   theme: "vs-dark",
-// });
+    monaco.editor.defineTheme('myTheme', {
```

Comment:
Lets not change the color of our own cursor and keep it default.
Lets only change the cursor characteristics of other users.